services/database.js

```diff
@@ -0,0 +1,20 @@
+const Url = require("../models/url");
+const shortid = require("shortid");
+
+const findUrl = async (originalUrl) => {
+  return await Url.findOne({ originalUrl });
```

Comment:
Errors are not handled here and error handling to the database operations can help catch and handle any errors that might occur.

The refactored code to include error handling can be like this 

```
const findUrl = async (originalUrl) => {
  try {
    return await Url.findOne({ originalUrl });
  } catch (err) {
    console.error(`Error occurred while finding URL`);
    throw err;
  }
};
```